src/System Application/App/Word Templates/src/WordTemplateImpl.Codeunit.al

```diff
@@ -609,6 +609,10 @@ codeunit 9988 "Word Template Impl."
         TempDocumentSharing.Insert();
 
         DocumentSharing.Share(TempDocumentSharing);
+
+        if TempDocumentSharing.State = TempDocumentSharing.State::Cancelled then
```

Comment:
Would be nice to cover also this one scenario in tests
\src\System Application\Test\Word Templates\src\*
I would say codeunit 130443 "Word Templates Test" is good fit for it